### DISCUSSION TITLE
Improve message "Not all code paths return a value" in case of entirely missing return statement

### DISCUSSION THREAD

**Discussion by GSPP** [dotnet/roslyn#59322] [Ideas]
Sometimes, you plainly forget to add any return statement at all. This results in the error

> Not all code paths return a value

which is correct but it sounds a bit cryptic. It sounds like there is some complicated code path that does not end in a return.

Would it be a good idea to detect when there is no return statement, and output a clearer message? Like this:

> Your function does not have any return statement, like, at all... Did you forget something?

**Comment by CyrusNajmabadi** [comment]
I don't really see how the second message is any clearer.

**Reply by RikkiGibson** [reply]
We've gotten feedback before from people who teach C# that "not all code paths return a value" could be better. I wish I could find the discussion, it essentially suggests adding some enhancement to the IDE that breaks down what the diagnostic means in more colloquial langauge.

**Reply by DimitarBogdanov** [reply]
Just a thought, this could be a possible message as well:
> Non-void method does not return a value.
